libraries/chain/include/eosio/chain/global_property_object.hpp

```diff
@@ -118,25 +129,77 @@ namespace eosio { namespace chain {
       chain_id_type                       chain_id;
       kv_database_config                  kv_configuration;
       wasm_config                         wasm_configuration;
+
+      static constexpr uint32_t minimum_version_with_extension = 6;
+
+      struct extension_v0 {
+         // libstdc++ requires the following two constructors to work. 
+         extension_v0(){};
+         extension_v0(block_num_type num, flat_set<account_name> participants)
+             : proposed_security_group_block_num(num)
+             , proposed_security_group_participants(std::move(participants)) {}
+
+         block_num_type                      proposed_security_group_block_num = 0;
+         flat_set<account_name>              proposed_security_group_participants;
+      };
+
+      // for future extensions, please use the following pattern:
+      // 
+      // struct extension_v1 : extension_v0 { new_field_t new_field; };
+      // using extension_t = std::variant<extension_v0, extension_v1>;
+      //  
+      // In addition, get_gpo_extension(), set_gpo_extension() and 
+      // eosio::ship_protocol::global_property from ship_protocol.hpp 
+      // in abieos has to be changed accordingly. 
+
+      using extension_t = std::variant<extension_v0>;
+      extension_t extension;
    };
 
+   inline snapshot_global_property_object::extension_t get_gpo_extension(const global_property_object& gpo) {
+      return snapshot_global_property_object::extension_v0{
+          gpo.proposed_security_group_block_num,
+          {gpo.proposed_security_group_participants.begin(), gpo.proposed_security_group_participants.end()}};
+   }
+
+   inline void set_gpo_extension(global_property_object&                        gpo,
+                                 snapshot_global_property_object::extension_t&& extension) {
```

Comment:
Why a rvalue reference instead of just const& ? It is only copied from, so seems like const& would be better.

Reply:
that was left over from the original change that didn't use shared_set in global_property_object and that was missed, cleaned up